nixos/modules/system/boot/systemd/tmpfiles.nix

```diff
@@ -100,6 +155,18 @@ in
           ${concatStringsSep "\n" cfg.rules}
         '';
       })
+      (pkgs.runCommand "systemd-extra-tmpfiles" {} ''
+        mkdir -p $out/lib/tmpfiles.d
+        cd $out/lib/tmpfiles.d
+
+        ${concatMapStrings (namespace:
+          concatMapStrings (path:
+            concatMapStrings (entry: ''
+              echo "${entry.type} ${path} ${entry.mode} ${entry.user} ${entry.group} ${entry.age} ${entry.argument}" >> nixos-${namespace}.conf
+            '') (attrValues cfg.settings.${namespace}.${path}))
+          (attrNames cfg.settings.${namespace})
+        ) (attrNames cfg.settings)}
+      '')
```

Comment:
This would be better as a `writeTextDir` rather than a `runCommand`.

And I'd prefer to see it written with `mapAttrsToList` rather than using `attrNames` and mapping with `foo.${name}` syntax.

Reply:
I can try to rework it to that. basically I just copied the code from above and extended it a bit until it worked.

Reply:
hmm, it seems writeTextDir creates only a single file. so we would have a derivation for each "namespace"/file. That doesn't seem like a lot of overhead but could be slower if we have a lot of namespaces

Reply:
I changed it now to use writeTextDir and mapAttrsToList